src/app/domain/authentication/services/SynchronizeService.ts

```diff
@@ -1,14 +1,51 @@
 import Minilog from '@cozy/minilog'
+import { getDeviceName } from 'react-native-device-info'
 
 import type CozyClient from 'cozy-client'
+import { getErrorMessage } from 'cozy-intent'
 
-const log = Minilog('SynchronizeService')
+import { SOFTWARE_NAME } from '/libs/constants'
+import { saveClient } from '/libs/clientHelpers/persistClient'
+import { safePromise } from '/utils/safePromise'
+
+export const syncLog = Minilog('📱 SynchronizeService')
 
 export const synchronizeDevice = async (client: CozyClient): Promise<void> => {
   try {
     await client.getStackClient().fetchJSON('POST', '/settings/synchronized')
-    log.info('📱 Device synchronized')
+    syncLog.info('Device synchronized')
+  } catch (error) {
+    syncLog.warn('Error while synchronizing device', error)
+  }
+}
+
+export const getClientName = async (): Promise<string> =>
+  `${SOFTWARE_NAME} (${await getDeviceName()})`
+
+export const checkClientName = async (client: CozyClient): Promise<void> => {
+  try {
+    const { clientName, ...oauthOptions } = client.getStackClient().oauthOptions
+    const newClientName = await getClientName()
+
+    if (!clientName.startsWith(newClientName)) {
+      syncLog.info(`Updating client name...`, {
+        oldClientName: clientName
+      })
+
+      const response = await client.getStackClient().updateInformation({
+        ...oauthOptions,
+        clientName: newClientName
+      })
+
+      await saveClient(client)
+
+      syncLog.info(`Client name updated with "${newClientName}"`, {
+        OAuthOptions: response
+      })
+    }
   } catch (error) {
-    log.warn('Error while synchronizing device', error)
+    syncLog.error('Failed to update clientName', getErrorMessage(error))
+  } finally {
+    safePromise(synchronizeDevice)(client)
```

Comment:
nit: I would pref that we call synchronizeDevice on app.js then getClientName is called by synchronizeDevice.

As a developper I don't expect that calling `checkClientName` would produce a synchronization. The reverse seems more logic.

Reply:
I agree. Do you think I should investigate with back-end team why running sync before update triggers an error in the stack? But that means we can't deploy this PR today

Reply:
Isn't it because both call are async and fire&forget so the stack will receive both call at the same time?

Reply:
https://github.com/cozy/cozy-stack/pull/4029 update on stack and I also refactored a bit so we always check if the name is correct before syncing when app starts